### DISCUSSION TITLE
JSON:API Compatibility

### DISCUSSION THREAD

**Discussion by szimre-bc** [spatie/laravel-data#42] [Q&A]
Is it possible to set up this package to produce valid JSON:API responses? The key issue for me here is that unlike the default Laravel JsonResource it omits the data wrapper when producing a simple response but uses it with paginated responses whereas the JSON:API specification says that

> ### [5.1 Top Level](https://jsonapi.org/format/#document-top-level)
>  A JSON object MUST be at the root of every JSON:API request and response containing data. This object defines a document’s “top level”.
>
>  A document MUST contain at least one of the following top-level members:
> - data: the document’s “primary data”
> - errors: an array of error objects
> - meta: a meta object that contains non-standard meta-information.

I would like to go from:
```
[
    {
        "name": "Never Gonna Give You Up",
        "artist": "Rick Astley"
    },
    {
        "name": "Giving Up on Love",
        "artist": "Rick Astley"
    }
]
```
to a JSON:API valid response:
```
{
    "data" : [
        {
            "name" : "Never Gonna Give You Up",
            "artist" : "Rick Astley"
        },
        {
            "name" : "Giving Up on Love",
            "artist" : "Rick Astley"
        }
    ]
}
```

Can't seem to figure out how to change this behaviour.
Thanks!

**Comment by rubenvanassche** [comment]
I've created a PR for this: https://github.com/spatie/laravel-data/pull/43 we're looking internally if this is the best solution for the package.
